### PR TITLE
Added more unit tests for offset and zookeeper.

### PR DESCRIPTION
**Pull request by lightswitch05**
Added more tests involving offset and zookeeper.

Coverage before new tests:

```
Statements   : 84.3% ( 1681/1994 )
Branches     : 67.07% ( 495/738 )
Functions    : 85.25% ( 393/461 )
Lines        : 85.58% ( 1626/1900 )
```

Coverage after new tests:

```
Statements   : 85.61% ( 1707/1994 )
Branches     : 68.7% ( 507/738 )
Functions    : 87.2% ( 402/461 )
Lines        : 87% ( 1653/1900 )
```

### CODE REVIEW

File: test/test.offset.js [hunk 1]
```diff
@@ -59,6 +59,16 @@ describe('Offset', function () {
         done(err);
       });
     });
+
+    it('should keeping calling fetch until offset is ready', function (done) {
+      var topic = '_exist_topic_3_test';
+      var topics = [ { topic: topic } ];
+      offset.ready = false;
+      offset.fetch(topics, function (err) {
```
Comment (hyperlink) [inline test/test.offset.js]: Could we just pass in `done` here?

File: test/test.offset.js [hunk 2]
```diff
@@ -71,6 +81,16 @@ describe('Offset', function () {
         done(err);
       });
     });
+
+    it('should keep calling commit until offset is ready', function (done) {
+      var topic = '_exist_topic_3_test';
+      var topics = [ { topic: topic, offset: 10 } ];
+      offset.ready = false;
+      offset.commit('_groupId_commit_test', topics, function (err) {
```
Comment (hyperlink) [inline test/test.offset.js]: Same as line 67.

**Comment by lightswitch05** [comment]
I updated the pull request per code review comments. Those were sloppy mistakes.

File: test/test.zookeeper.js [hunk 3]
```diff
@@ -234,4 +250,30 @@ describe('Zookeeper', function () {
...
+
+  describe('#deletePartitionOwnership', function () {
+    it('it removes ownership', function (done) {
+      var groupId = 'myGroup' + randomId();
+      var consumerId = 'myConsumer' + randomId();
+      var topic = '_exist_topic_3_test';
+
+      zk.addPartitionOwnership(consumerId, groupId, topic, 0, function (error) {
+        if (error) {
+          done(error);
```
Comment (hyperlink) [inline test/test.zookeeper.js]: Probably should add a `return` on this line.

Comment (lightswitch05) [inline test/test.zookeeper.js]: done

**Comment by hyperlink** [comment]
LGTM. Thanks @lightswitch05

**Pull request merged**